examples/if_cuda/BUILD.bazel

```diff
@@ -1,9 +1,10 @@
-load("@rules_cuda//cuda:defs.bzl", "cuda_library")
+load("@rules_cuda//cuda:defs.bzl", "cuda_library", "requires_cuda")
 
 cuda_library(
     name = "kernel",
     srcs = ["kernel.cu"],
     hdrs = ["kernel.h"],
+    target_compatible_with = requires_cuda(),
```

Comment:
Just a side note: users could still get non-so-obvious errors from bazel in case the target compatibility is not set.